BuildScript/BuildTasks/TestConfigurationsFilter.cs

```diff
@@ -0,0 +1,269 @@
+    {
+      foreach (var testConfiguration in testConfigurations)
+      {
+        if (!IsSupportedBrowser (testConfiguration.Browser))
+          yield return GetErrorMessageForUnsupportedFilterValue (TestConfigurationMetadata.Browser, testConfiguration.Browser);
+
+        if (!IsSupportedPlatform (testConfiguration.Platform))
+          yield return GetErrorMessageForUnsupportedFilterValue (TestConfigurationMetadata.Platform, testConfiguration.Platform);
+
+        if (!IsSupportedDatabaseSystem (testConfiguration.DatabaseSystem))
+          yield return GetErrorMessageForUnsupportedFilterValue (TestConfigurationMetadata.DatabaseSystem, testConfiguration.DatabaseSystem);
+
+        if (!IsSupportedExecutionRuntime (testConfiguration.ExecutionRuntime))
+          yield return GetErrorMessageForUnsupportedFilterValue (TestConfigurationMetadata.ExecutionRuntime, testConfiguration.ExecutionRuntime);
+
+        if (!IsSupportedTargetRuntime (testConfiguration.TargetRuntime))
+          yield return GetErrorMessageForUnsupportedFilterValue (TestConfigurationMetadata.TargetRuntime, testConfiguration.TargetRuntime);
+
+        if (!IsSupportedConfigurationID (testConfiguration.ConfigurationID))
+          yield return GetErrorMessageForUnsupportedFilterValue (TestConfigurationMetadata.ConfigurationID, testConfiguration.TargetRuntime);
+      }
+    }
+
+    private IEnumerable<string> GetErrorsForEmptyEnabledMetadata ()
+    {
+      if (!_enabledBrowsers.Any())
+        yield return GetErrorMessageForMissingEnabledValues (TestConfigurationMetadata.Browser);
+
+      if (!_enabledDatabaseSystems.Any())
+        yield return GetErrorMessageForMissingEnabledValues (TestConfigurationMetadata.DatabaseSystem);
+
+      if (!_enabledExecutionRuntimes.Any())
+        yield return GetErrorMessageForMissingEnabledValues (TestConfigurationMetadata.ExecutionRuntime);
+
+      if (!_enabledPlatforms.Any())
+        yield return GetErrorMessageForMissingEnabledValues (TestConfigurationMetadata.Platform);
+
+      if (!_enabledTargetRuntimes.Any())
+        yield return GetErrorMessageForMissingEnabledValues (TestConfigurationMetadata.TargetRuntime);
+    }
+
+    private string GetErrorMessageForUnsupportedFilterValue (string metadata, string value)
+    {
+      return $"Cannot filter with unsupported {metadata} '{value}'";
+    }
+
+    private string GetErrorMessageForMissingEnabledValues (string metadata)
+    {
+      return $"No enabled {metadata}s were provided.";
+    }
+
+    private bool IsSupportedPlatform (string platform)
+    {
+      return _supportedPlatforms.Contains (platform, StringComparer.OrdinalIgnoreCase);
+    }
+
+    private bool IsSupportedDatabaseSystem (string database)
+    {
+      if (string.Equals (database, MetadataValueConstants.NoDB, StringComparison.OrdinalIgnoreCase))
+        return true;
+
+      return _supportedDatabaseSystems.Contains (database, StringComparer.OrdinalIgnoreCase);
```

Comment:
Why don't we add NoDB, NoBrowser, etc to the _supportedXxx collections? Either inside the filter or already at the callsite?